core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -163,6 +163,15 @@ sealed trait ZValidation[+W, +E, +A] { self =>
       case Success(w, a) => Success(f(w), a)
     }
 
+  final def or[E1, W1 >: W, A1 >: A](that: ZValidation[W1, E1, A1])(implicit ev: E <:< W1): ZValidation[W1, E1, A1] =
+    orWith[W1, E1, A1](_.map(ev))(that)
+
+  final def orWith[W1 >: W, E1, A1 >: A](convert: Chunk[E] => Chunk[W1])(that: ZValidation[W1, E1, A1]): ZValidation[W1, E1, A1] =
+    self match {
+      case Failure(log, errors) => that.mapLogAll(log ++ convert(errors) ++ _)
```

Comment:
The semantics here are not very well defined. We now have three fairly different things that we are putting into the same collection here: (1) the log from the original value that failed, (2) the errors from the original value that failed, and (3) the log from the new value that failed.